### ISSUE TITLE
Bitbucket Private Repos

### ISSUE THREAD

**Issue by bradmkjr** [afragen/github-updater#175]
Currently in both Dev and Master branch, it appears there is a logical IF statement which fails to set the correct value when setting up headers for a bitbucket api request.

Bitbucket_API.php around line 314

```php
		/**
		 * Check whether attempting to update private Bitbucket repo.
		 */
		if ( isset( $this->type->repo ) &&
			! empty( parent::$options[ $this->type->repo ] ) &&
		     false !== strpos( $url, $this->type->repo )
		) {
			$bitbucket_private = true;
		}
```
This didn't set the flag correctly for me for a private repo. I was able to install a private repo, but it was returning a 401 from Bitbucket when I checked for updates. If I overrode the if statement, and manually set the flag to true it worked perfectly.

**Comment by afragen** [comment]
Have you checked the setting for the private repo in the Settings page?

**Comment by bradmkjr** [comment]
You are correct. I set my username and password. I installed a private repo Bitbucket plugin, but I didn't return to the settings page to confirm it was a private repo before attempting to update. I think many people may make this same mistake, as it isn't logical to need to check the box again after checking it during the install. 

Perhaps this could be changed to automatically be checked if using the installer for a private repo?

Thanks for the super fast reply!
